src/main/java/com/epam/internship/carrental/car/CarController.java

```diff
@@ -0,0 +1,294 @@
+     *
+     * @param carService service which provides access to the service layer.
+     */
+    @Autowired
+    public CarController(final CarServiceImpl carService) {
+        this.carService = carService;
+    }
+
+    /**
+     * Adds a new Car to the database, where the Car is defined in the request parameters.
+     * <pre>
+     * Method GET
+     * URL /api/v1/add
+     * </pre>
+     * Sample call: /api/v1/add?make=Dacia&model=1310&carType=Sedan&seats=5&fuelUsage=12.7&carGearbox=Manual
+     *
+     * @param make       maker of the car to be added.
+     * @param model      model of the car to be added.
+     * @param carType    carType of the car to be added.
+     * @param seats      number of seats in the car to be added.
+     * @param fuelUsage  fuel usage in liter/100km of the car to be added.
+     * @param carGearbox gearbox of the car to be added.
+     * @return ResponseEntity with Response Code 200 on success.
+     */
+    @GetMapping(path = "/add")
+    public @ResponseBody
+    ResponseEntity addNewCar(@RequestParam final String make,
+                             @RequestParam final String model,
+                             @RequestParam final CarType carType,
+                             @RequestParam final int seats,
+                             @RequestParam final double fuelUsage,
+                             @RequestParam final CarGearbox carGearbox) {
+        return carService.addNewCar(make, model, carType,
+                seats, fuelUsage, carGearbox);
+    }
+
+    /**
+     * Adds a new Car to the database. The car is defined in the body of the request in JSON format.
+     * <pre>
+     * Method POST
+     * URL /api/v1/add
+     * </pre>
+     * Sample call /api/v1/add
+     *
+     * @param car car to be added
+     * @return ResponseEntity with Response Code 200 on success.
+     */
+    @PostMapping(path = "/add", consumes = "application/json")
+    public @ResponseBody
+    ResponseEntity addNewCar(@RequestBody final Car car) {
+        return carService.addNewCar(car);
+    }
+
+    /**
+     * Used to get all cars by a given maker.
+     * <pre>
+     * Method GET
+     * URL /api/v1/search
+     * </pre>
+     * Sample call /api/v1/add?make=Dacia
+     *
+     * @param make maker of cars to be retrieved
+     * @return ResponseEntity which contains an Iterable list of cars made by the maker,
+     * and Response Code 200 on success.
+     */
+    @GetMapping(path = "/search")
+    public @ResponseBody
+    ResponseEntity<Iterable<Car>> getCarsByMake(@RequestParam final String make) {
+        return carService.getCarsByMake(make);
+    }
+
+    /**
+     * Gets all car from the database.
+     * <pre>
+     * Method GET
+     * URL /api/v1/all
+     * </pre>
+     * Sample call /api/v1/all
+     *
+     * @return ResponseEntity which contains an Iterable list of all cars,
+     * and Response Code 200 on success.
+     */
+    @GetMapping(path = "/all")
```

Comment:
The name of this service is ambiguous. If you want to show all the available cars, then the service should be named like GET `/api/v1/car`. Also, I do not suggest isolating this call from the `search` service - the search exactly serves this purpose. The search service without parameters should return all the cars. That said, I suggest removing this method here.